### DISCUSSION TITLE
Unexpected Behavior When Surrounding Text Using Visual Mode

### DISCUSSION THREAD

**Discussion by eeriemyxi** [kylechui/nvim-surround#280] [Q&A]
This is my config:
```lua
return {
    "kylechui/nvim-surround",
    version = "*",
    event = "VeryLazy",
    config = function()
        require("nvim-surround").setup({
            keymaps = {
                insert = "<C-g>s",
                insert_line = "<C-g>S",
                normal = "ys",
                normal_cur = "yss",
                normal_line = "yS",
                normal_cur_line = "ySS",
                visual = "k",
                visual_line = "k",
                delete = "ds",
                change = "cs",
                change_line = "cS",
            },
        })
    end,
}
```
If my cursor is at:
```
word_here
^
```
Then If I do `vwk(` or `vwk)`, the result I get is:
```
(
word_here
)
```
I am not sure if I should be expecting that. I expect the resultant to be `(word_here)`. This happens even if I try to surround on a single character.

**Comment by eeriemyxi** [comment]
Any help?

**Reply by kylechui** [reply]
Sorry I didn't see this. You have set both `visual` and `visual_line` to use
the same keymap, which is causing a conflict. In this case, `visual_line` is
overriding `visual`. You can fix this by setting them to different keymaps.
